File: TodoApi/Models/Drink.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace TodoApi.Models
{
    /*
    public class TodoItem
    {
        public long Id { get; set; }

        public string Name { get; set; }

        public bool IsComplete { get; set; }
    }
    */
    public class DrinkGeneral
    {
        public string strDrink { get; set; }
        public string strDrinkThumb { get; set; }
        public string idDrink { get; set; }

        public long Id {get; set; }
    }

    public class DrinkResult
    {
        public List<DrinkGeneral> drinks { get; set; }
    }

    // Root myDeserializedClass = JsonConvert.DeserializeObject<Root>(myJsonResponse); 
    public class DrinkDetails
    {
        public string idDrink { get; set; }
        public string strDrink { get; set; }
        public object strDrinkAlternate { get; set; }
        public object strDrinkES { get; set; }
        public object strDrinkDE { get; set; }
        public object strDrinkFR { get; set; }
        [JsonProperty("strDrinkZH-HANS")]
        public object StrDrinkZHHANS { get; set; }
        [JsonProperty("strDrinkZH-HANT")]
        public object StrDrinkZHHANT { get; set; }
        public object strTags { get; set; }
        public object strVideo { get; set; }
        public string strCategory { get; set; }
        public object strIBA { get; set; }
        public string strAlcoholic { get; set; }
        public string strGlass { get; set; }
        public string strInstructions { get; set; }
        public object strInstructionsES { get; set; }
        public string strInstructionsDE { get; set; }
        public object strInstructionsFR { get; set; }
        [JsonProperty("strInstructionsZH-HANS")]
        public object StrInstructionsZHHANS { get; set; }
        [JsonProperty("strInstructionsZH-HANT")]
        public object StrInstructionsZHHANT { get; set; }
        public string strDrinkThumb { get; set; }
        public string strIngredient1 { get; set; }
        public string strIngredient2 { get; set; }
        public string strIngredient3 { get; set; }
        public string strIngredient4 { get; set; }
        public string strIngredient5 { get; set; }
        public object strIngredient6 { get; set; }
        public object strIngredient7 { get; set; }
        public object strIngredient8 { get; set; }
        public object strIngredient9 { get; set; }
        public object strIngredient10 { get; set; }
        public object strIngredient11 { get; set; }
        public object strIngredient12 { get; set; }
        public object strIngredient13 { get; set; }
        public object strIngredient14 { get; set; }
        public object strIngredient15 { get; set; }
        public string strMeasure1 { get; set; }
        public string strMeasure2 { get; set; }
        public string strMeasure3 { get; set; }
        public string strMeasure4 { get; set; }
        public string strMeasure5 { get; set; }
        public object strMeasure6 { get; set; }
        public object strMeasure7 { get; set; }
        public object strMeasure8 { get; set; }
        public object strMeasure9 { get; set; }
        public object strMeasure10 { get; set; }
        public object strMeasure11 { get; set; }
        public object strMeasure12 { get; set; }
        public object strMeasure13 { get; set; }
        public object strMeasure14 { get; set; }
        public object strMeasure15 { get; set; }
        public object strImageSource { get; set; }
        public object strImageAttribution { get; set; }
        public string strCreativeCommonsConfirmed { get; set; }
        public string dateModified { get; set; }
    }

    public class DrinkDetailResult
    {
        public List<DrinkDetails> drinkData { get; set; }
    }
}

File: drink-front/src/components/drink.js
import React from 'react';
import './drink.css';

const Drink = ({drinkData}) => {
  

  return (
    <div className="drinkContainer">
      <div className="drinkHeroContainer">
        <img className="drinkImg" src={drinkData.strDrinkThumb} />
        <h1>{drinkData.strDrink}</h1>
      </div>
      <div className="detailsContainer">
        <div className="row left">

          <h3 className="instructionsHeading">Instructions</h3>
          <p className="instructions">{drinkData.strInstructions}</p>
        </div>

        <div className="row right">
          <h3>Ingredients</h3>
          <ul>
              { (drinkData.strIngredient1 !== null) ? <li><b>{drinkData.strIngredient1}</b> <span>{drinkData.strMeasure1}</span></li> : null}
              { (drinkData.strIngredient2 !== null) ? <li><b>{drinkData.strIngredient2}</b> <span>{drinkData.strMeasure2}</span></li> : null}
              { (drinkData.strIngredient3 !== null) ? <li><b>{drinkData.strIngredient3}</b> <span>{drinkData.strMeasure3}</span></li> : null}
              { (drinkData.strIngredient4 !== null) ? <li><b>{drinkData.strIngredient4}</b> <span>{drinkData.strMeasure4}</span></li> : null}
              { (drinkData.strIngredient5 !== null) ? <li><b>{drinkData.strIngredient5}</b> <span>{drinkData.strMeasure5}</span></li> : null}
              { (drinkData.strIngredient6 !== null) ? <li><b>{drinkData.strIngredient6}</b> <span>{drinkData.strMeasure6}</span></li> : null}
          </ul>
        </div>

      </div>
    </div>
  )
}


export default Drink;